autogpt_platform/backend/migrations/20250313043016_update_store_submissions_format/migration.sql

```diff
@@ -0,0 +1,326 @@
+/*
+  Warnings:
+
+  - The values [DAFT] on the enum `SubmissionStatus` will be removed. If these variants are still used in the database, this will fail.
+  - You are about to drop the column `isApproved` on the `StoreListing` table. All the data in the column will be lost.
+  - You are about to drop the column `slug` on the `StoreListingVersion` table. All the data in the column will be lost.
+  - You are about to drop the `StoreListingSubmission` table. If the table is not empty, all the data it contains will be lost.
+  - A unique constraint covering the columns `[activeVersionId]` on the table `StoreListing` will be added. If there are existing duplicate values, this will fail.
+  - A unique constraint covering the columns `[storeListingId,version]` on the table `StoreListingVersion` will be added. If there are existing duplicate values, this will fail.
+  - Made the column `storeListingId` on table `StoreListingVersion` required. This step will fail if there are existing NULL values in that column.
+
+*/
+
+-- First, drop all views that depend on the columns we're modifying
+DROP VIEW IF EXISTS platform."StoreSubmission";
+DROP VIEW IF EXISTS platform."StoreAgent";
+DROP VIEW IF EXISTS platform."Creator";
+
+-- Then proceed with the table changes
+BEGIN;
+-- Create the new enum type
+CREATE TYPE "SubmissionStatus_new" AS ENUM ('DRAFT', 'PENDING', 'APPROVED', 'REJECTED');
+
+-- Modify the column with the correct casing (Status with capital S)
+ALTER TABLE "StoreListingSubmission" ALTER COLUMN "Status" DROP DEFAULT;
+ALTER TABLE "StoreListingSubmission" ALTER COLUMN "Status" TYPE "SubmissionStatus_new" 
+    USING (CASE WHEN "Status"::text = 'DAFT' THEN 'DRAFT'::text 
+          ELSE "Status"::text 
+          END)::"SubmissionStatus_new";
+
+-- Rename the enum types
+ALTER TYPE "SubmissionStatus" RENAME TO "SubmissionStatus_old";
+ALTER TYPE "SubmissionStatus_new" RENAME TO "SubmissionStatus";
+DROP TYPE "SubmissionStatus_old";
+
+-- Set default back
+ALTER TABLE "StoreListingSubmission" ALTER COLUMN "Status" SET DEFAULT 'PENDING';
+COMMIT;
+
+-- Drop constraints
+ALTER TABLE "StoreListingSubmission" DROP CONSTRAINT IF EXISTS "StoreListingSubmission_reviewerId_fkey";
+
+-- Drop indexes
+DROP INDEX IF EXISTS "StoreListing_isDeleted_isApproved_idx";
+DROP INDEX IF EXISTS "StoreListingSubmission_storeListingVersionId_key";
+
+-- Modify StoreListing
+ALTER TABLE "StoreListing" 
+DROP COLUMN IF EXISTS "isApproved",
+ADD COLUMN IF NOT EXISTS "activeVersionId" TEXT,
+ADD COLUMN IF NOT EXISTS "hasApprovedVersion" BOOLEAN NOT NULL DEFAULT false,
```

Comment:
Shouldn't hasApprovedVersion will be equals to isApproved ? can we just alter it ?

Reply:
unfortunately isApproved adn has approved version should be the same thing but the db is currently in a weird state so its better to recaculate it imo

Reply:
where do we recalculate this? I don't see it calculated anywhere.

Reply:
I"m adding it:
https://github.com/Significant-Gravitas/AutoGPT/pull/9628/commits/477e82173e9a7d7664808496f715e4677def333d

let me know if this looks good